### PR TITLE
[BIOMAGE-1793] - Add sandbox ID to RDS cluster endpoint

### PR DESCRIPTION
**Pull request by aerlaut**
# Description
#### Problem
We want to have a separate staging RDS deployment so we can test changes to the database in isolation.

#### Solution
[IAC PR 326](https://github.com/hms-dbmi-cellenics/iac/pull/326) added sandbox id to deployed RDS instances. The API now needs to correctly connect to this instance. The changes made in this PR:
- introduce sandboxing, so the API can connect to the correct instance.
- Add a variable in `flux.yaml` to hold the RDS endpoint for this instance. The value of this endpoint would be filled in when staging an environment using `biomage stage`. Staging an RDS instance (`biomage stage --with_rds`) would fill the value with the rds cluster ID to connect to.
- Make the API use the cluster id to connect to the correct RDS instance.

# Details
#### URL to issue
https://biomage.atlassian.net/browse/BIOMAGE-1793

https://ui-agi-shabby-swan.scp-staging.biomage.net/
#### Link to staging deployment URL (or set N/A)
N/A

#### Links to any PRs or resources related to this PR
- IAC - https://github.com/hms-dbmi-cellenics/iac/pull/326
- API - https://github.com/hms-dbmi-cellenics/api/pull/323
- Biomage utils - https://github.com/hms-dbmi-cellenics/biomage-utils/pull/81

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [ ] Migrated any selector / reducer used to the new format.

### Manual/unit testing
- [X] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [X] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
- [X] Unit tests written **or** no unit tests required for change, e.g. documentation update.

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.

- [ ] Started end-to-end tests on the latest commit.

### Documentation updates
- [X] Relevant Github READMEs updated **or** no GitHub README updates required.
- [X] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Staging environment is unstaged before merging.
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -245,6 +247,7 @@ jobs:
 
           if [ "${{ matrix.environment }}" = "staging" ]; then
             SANDBOX_ID="STAGING_SANDBOX_ID"
+            RDS_CLUSTER_ID="STAGING_RDS_CLUSTER_ID"
```
Comment (cosa65) [inline .github/workflows/ci.yaml]: `STAGING_RDS_CLUSTER_ID` is brought from the changes in iac?

File: src/config/default-config.js [hunk 2]
```diff
@@ -63,6 +63,7 @@ const config = {
   port: parseInt(process.env.PORT, 10) || 3000,
   clusterEnv: process.env.CLUSTER_ENV || 'development',
   sandboxId: process.env.SANDBOX_ID || 'default',
+  rdsClusterId: process.env.RDS_CLUSTER_ID || 'default',
```
Comment (cosa65) [inline src/config/default-config.js]: `RDS_CLUSTER_ID` is not the clusterId, it's actually the sandboxId but passed through the flag of stage an rds or not. I would suggest either passing the flag for stage an rds or not (so a true or false) or rename this to `RDS_SANDBOX_ID` or something like that.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (aerlaut) [reply .github/workflows/ci.yaml]: No, it's just a placeholder that'll be changed to the correct value when staging by biomage-utils

File: src/config/default-config.js (hunk 2 above)
Comment (aerlaut) [reply src/config/default-config.js]: changed

**Review by cosa65 — APPROVED** [review]

**Pull request merged**